### ISSUE TITLE
Allow creating a translated message type programtically

### ISSUE THREAD

**Issue by amitaibu** [Gizra/message#91]
Like inD7:

``` php
    $message_type = message_type_create('foo');
    $message_type->{$property} = array(
      'en' => array(
        0 => array('value' => 'english text'),
      ),
      'l0' => array(
        0 => array('value' => $text1),
      ),
      'l1' => array(
        0 => array('value' => $text2),
      ),
    );
    $message_type->save();
```

**Comment by bojanz** [comment]
Since message types are config entities, doesn't it make sense to simply document how it's done there?
(By creating a config collection for every translation)

Address example: https://github.com/bojanz/address/blob/8.x-1.x/src/AddressFormatImporter.php#L99

**Comment by amitaibu** [comment]
If I understand you correctly, my plan is indeed to change the way it works, and have a config per translation (i.e. also the default one).

`MessageType::create()` should do the heavy lifting.

@bojanz , I know you are not doing anything now, and are completely out of touch of PHP & Drupal 8, so if you want to give this one a shot  -- I'm sure practicing contribution could help your career :wink:

**Comment by bojanz** [comment]
:)

I don't think you should change the way this works, there's no need. You can create a Translate operation/link/tab to show it right next to Edit, and the rest you get from Drupal for free. 
Furthermore, changing the behavior of a config entity so that it applies to multiple languages will go against expectations of people familiar with D8.

EDIT: Okay, maybe you just want MessageType::create() to act as a helper, and not change anything else. In that case I'd advise to add a new helper to MessageType, like createTranslation(). No need to change the semantics of the core method.

**Comment by amitaibu** [comment]
> MessageType::create() to act as a helper

Exactly. I'm thinking that `MessageType::create()` could be a wrapper to `MessageType::createTranslation()`

**Comment by RoySegall** [comment]
@amitaibu Do you want that maybe @HelenaEksler take this one?

**Comment by amitaibu** [comment]
sure.

**Comment by jhedstrom** [comment]
I don't fully understand why this is needed. Doesn't the core config translation module handle this?

**Comment by amitaibu** [comment]
The need is for a nicer DX for creating translated Messages. But maybe I'm missing as-well. Can you show how you would create a message in English and Spanish.
